### PR TITLE
Feat: Landing page

### LINKED ISSUE #174: Margarita Landing Page
- [x] header
- [x] headline
- [x] graphql playground
- [x] showcase & info
- [x] demo
- [x] footer

### PR DESCRIPTION
**Pull request by KarinaDavtyan**
Summary:
+ Added initial layout for landing page

+ Responsiveness will be improved

+ The design will be changed

Closes #174

### CODE REVIEW

File: .env-sample [hunk 1]
```diff
@@ -1 +0,0 @@
-GRAPHQL_URL=
```
Comment (tbergq) [inline .env-sample]: why is this file deleted?

File: apps/graphql/.env-example [hunk 2]
```diff
@@ -1,3 +0,0 @@
-API_KEY=
```
Comment (tbergq) [inline apps/graphql/.env-example]: why is it deleted?

File: apps/landingPage/package.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "gatsby-starter-default",
```
Comment (tbergq) [inline apps/landingPage/package.json]: maybe we should name this package differently?

File: apps/landingPage/package.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    "gatsby-plugin-offline": "^2.0.22",
+    "gatsby-plugin-react-helmet": "^3.0.2",
+    "gatsby-plugin-sharp": "^2.0.19",
+    "gatsby-source-filesystem": "^2.0.8",
+    "gatsby-transformer-sharp": "^2.1.13",
+    "prop-types": "^15.6.2",
+    "react": "^16.6.3",
+    "react-dom": "^16.6.3",
+    "react-helmet": "^5.2.0",
+    "styled-components": "^4.1.3"
```
Comment (tbergq) [inline apps/landingPage/package.json]: could you move dependencies that are already included in other `package.json` to peerDependency? like `@kiwicom/orbit-components` and `react`

File: apps/landingPage/package.json (hunk 3 above)
Comment (KarinaDavtyan) [reply apps/landingPage/package.json]: Just changed with commit

File: .env-sample (hunk 1 above)
Comment (KarinaDavtyan) [reply .env-sample]: Sorry misread initial configuration instructions, renamed files without doing a copy first

I added them back in a new commit

File: apps/graphql/.env-example (hunk 2 above)
Comment (KarinaDavtyan) [reply apps/graphql/.env-example]: Same⬆️

File: apps/landingPage/package-lock.json [hunk 5]
```diff
@@ -0,0 +1,15743 @@
+{
```
Comment (tbergq) [inline apps/landingPage/package-lock.json]: could we stick to having `yarn.lock` only please. No package-lock.json 😊

File: apps/landingPage/package.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "Margarita's landing page",
```
Comment (tbergq) [inline apps/landingPage/package.json]: Sorry, I should have mentioned in latest commit,  but can we keep naming convention. `@kiwicom/margarita-landing-page`

File: apps/landingPage/package.json (hunk 4 above)
Comment (KarinaDavtyan) [reply apps/landingPage/package.json]: Moved in the latest commit

File: apps/landingPage/package.json (hunk 6 above)
Comment (KarinaDavtyan) [reply apps/landingPage/package.json]: Will fix now, and should i keep description blanked as it is now?

File: apps/landingPage/package-lock.json (hunk 5 above)
Comment (KarinaDavtyan) [reply apps/landingPage/package-lock.json]: Ok! never again

File: apps/landingPage/src/components/demo.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      </Text>
+      </div>
+    </div>
+    <Text size="large" italic={true}>
+      If you don't like <a href="https://tequila.kiwi.com/portal/login" style={{ color: defaultTokens.colorTextPrimary }}>Tequila</a> alone, grab a <Link to='/' style={{ color: defaultTokens.colorTextPrimary }}>Margarita</Link>
+    </Text>
+  </div>
+)
+
+const styles = {
```
Comment (tbergq) [inline apps/landingPage/src/components/demo.js]: Also I see `styled-components` as one of the dependencies. Could we use that for styling instead of inline-styles?

Comment (KarinaDavtyan) [reply apps/landingPage/src/components/demo.js]: Ok! Changed all inline styles into styled components

File: apps/landingPage/src/styles/global.css [hunk 8]
```diff
@@ -0,0 +1,4 @@
+body {
+  margin: 0;
+  -webkit-font-smoothing: antialiased;
+}
```
Comment (tbergq) [inline apps/landingPage/src/styles/global.css]: Also this should be replace with [styled-components](https://www.styled-components.com/docs/api#createglobalstyle)

File: apps/landingPage/src/components/demo.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+import React from 'react';
+import { Text, Heading } from '@kiwicom/orbit-components/lib/';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { Link } from 'gatsby';
+import styled from 'styled-components';
+
+type Props = {
+  iframeWidth: number,
+  iframeHeight: number,
```
Comment (tbergq) [inline apps/landingPage/src/components/demo.js]: Could this be exact type with read-only props?
```
type Props = {|
  +iframeWidth: number,
  +iframeHeight: number,
|};
```

File: apps/landingPage/src/components/headline.js [hunk 10]
```diff
@@ -0,0 +1,82 @@
+// @flow
+
+import React from 'react';
+import { Heading, Text } from '@kiwicom/orbit-components/lib/';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import styled from 'styled-components';
+
+type Props = {
+  iframeWidth: number,
+};
```
Comment (tbergq) [inline apps/landingPage/src/components/headline.js]: also please exact and read-only 😊

File: apps/landingPage/src/components/info.js [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+// $FlowFixMe - Cannot resolve module RelativeImageStub ( possible solution https://bit.ly/2BYfHRr )
+import ios from '../images/ios.png';
+// $FlowFixMe - Cannot resolve module RelativeImageStub ( possible solution https://bit.ly/2BYfHRr )
+import android from '../images/android.png';
+// $FlowFixMe - Cannot resolve module RelativeImageStub ( possible solution https://bit.ly/2BYfHRr )
+import web from '../images/web.png';
+
+type Props = {
+  iframeWidth: number,
+};
```
Comment (tbergq) [inline apps/landingPage/src/components/info.js]: read-only and exact

File: apps/landingPage/src/styles/global.css (hunk 8 above)
Comment (KarinaDavtyan) [reply apps/landingPage/src/styles/global.css]: Sorry missed that, fixed now

**Comment by RobinCsl** [comment]
Once it's ready, let me know and we can have it on auto-deploy on Netlify too.

File: apps/landingPage/src/components/footer.js [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+  </Footer>
+);
+
+const Footer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin-top: 3vh;
+  height: 10vh;
+  align-items: center;
+  border-top: 2px solid #f5f7f9;
```
[the remark this answers is not in the document]
Comment (KarinaDavtyan) [reply apps/landingPage/src/components/footer.js]: Changed to separator

**Comment by tbergq** [comment]
LGTM, but I think also @RobinCsl could review it 😊 

Also please squash your commits and rebase with master.

File: apps/landingPage/gatsby-config.js [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+        start_url: `/`,
+        background_color: `#663399`,
+        theme_color: `#663399`,
+        display: `minimal-ui`,
+        icon: `src/images/logo_margarita.png`, // This path is relative to the root of the site.
+      },
+    },
+    // this (optional) plugin enables Progressive Web App + Offline functionality
+    // To learn more, visit: https://gatsby.app/offline
+    // 'gatsby-plugin-offline',
```
Comment (FilipMessa) [inline apps/landingPage/gatsby-config.js]: Do we need this comment?

File: apps/landingPage/package.json [hunk 14]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "@kiwicom/margarita-landing-page",
+  "private": true,
+  "version": "0.1.0",
+  "dependencies": {
```
Comment (FilipMessa) [inline apps/landingPage/package.json]: I realized that there are few gatsby plugins which are not used anywhere, can you check and clean them.

File: apps/landingPage/package.json [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    "gatsby": "^2.0.105",
+    "gatsby-image": "^2.0.20",
+    "gatsby-plugin-flow": "^1.0.2",
+    "gatsby-plugin-manifest": "^2.0.15",
+    "gatsby-plugin-offline": "^2.0.22",
+    "gatsby-plugin-react-helmet": "^3.0.2",
+    "gatsby-plugin-sharp": "^2.0.19",
+    "gatsby-source-filesystem": "^2.0.8",
+    "gatsby-transformer-sharp": "^2.1.13",
+    "prop-types": "^15.6.2",
```
Comment (FilipMessa) [inline apps/landingPage/package.json]: I think this package also does not use anywhere.

**Comment by FilipMessa** [comment]
What was the reason to remove `yarn.lock`?

File: apps/landingPage/package.json (hunk 15 above)
Comment (KarinaDavtyan) [reply apps/landingPage/package.json]: There were too many conflicts in yarn.lock

Comment (KarinaDavtyan) [reply apps/landingPage/package.json]: which one? prop-types?

File: apps/landingPage/package.json [hunk 16]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@kiwicom/margarita-landing-page",
+  "private": true,
+  "version": "0.1.0",
+  "dependencies": {
+    "@kiwicom/orbit-components": "^0.28.2",
+    "@kiwicom/orbit-design-tokens": "^0.3.0",
+    "gatsby": "^2.0.105",
+    "gatsby-image": "^2.0.20",
```
Comment (FilipMessa) [inline apps/landingPage/package.json]: The `gatsby-image` is nowhere to be used. Or maybe I did not find where?

Comment (KarinaDavtyan) [reply apps/landingPage/package.json]: You are right, its also not used

**Comment by FilipMessa** [comment]
BTW maybe will be nice use `gatsby-image` in future because of the images optimization, but yes it isn't part of this PR.

https://medium.com/@kyle.robert.gill/ridiculously-easy-image-optimization-with-gatsby-js-59d48e15db6e

**Comment by michalsanger** [comment]
How can I run it? The readme looks like default Gatsby (plz fix). 

I tried `yarn gatsby develop` but that does not work in root or `apps/landingPage`

**Comment by KarinaDavtyan** [comment]
> How can I run it? The readme looks like default Gatsby (plz fix).
> 
> I tried `yarn gatsby develop` but that does not work in root or `apps/landingPage`

Fixed now.
Added `yarn landing-page` from the root folder
and `yarn start` from apps/landingPage
Also fixed readme

File: apps/landingPage/src/components/footer.js [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+  Facebook,
+  Twitter,
+} from '@kiwicom/orbit-components/lib/icons/';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import styled from 'styled-components';
+
+// $FlowFixMe - Cannot resolve module RelativeImageStub ( possible solution https://bit.ly/2BYfHRr )
+import logo from '../images/logo.png';
+
+const links = [
```
Comment (RobinCsl) [inline apps/landingPage/src/components/footer.js]: These could be extract to a config file in the future, because I believe we will re-use these links throughout the page.
It's just a minor improvement, and could be done later.

File: apps/landingPage/src/components/header.js [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+];
+
+export default () => (
+  <Header>
+    <a href="https://code.kiwi.com/">
+      <Logo src={logo} alt="Logo" />
+    </a>
+    <HeaderRight>
+      {links.map(el => {
+        const internal = /^\/(?!\/)/.test(el.link);
```
Comment (RobinCsl) [inline apps/landingPage/src/components/header.js]: Is that necessary? Does Gatsby Link only work for internal links?

File: apps/landingPage/src/components/headline.js [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+`;
+
+const PlaneImage = styled.div`
+  width: 76vw;
+  padding: 8vh 0 6vh 0;
+  justify-content: center;
+  display: flex;
+  height: 40vh;
+  margin-bottom: 4vh;
+  background-image: url(https://tequila.kiwi.com/portal/static/media/airport-runway.4acb8968.jpg);
```
Comment (RobinCsl) [inline apps/landingPage/src/components/headline.js]: Are we sure this hash in the filename will never change? That might break the site. One solution could be to extract it and add it to the repo. Although I am not sure this the best approach either.
@tbergq thoughts?

File: apps/landingPage/src/components/info.js [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+      </ContainerShowcase>
+    </ContainerTop>
+    <ContainerInfo>
+      <ContainerHeaderBottom>
+        <Heading>It has these new features</Heading>
+      </ContainerHeaderBottom>
+      <ContainerInfoBottom width={iframeWidth}>
+        <Info>
+          <Text>
+            {' '}
```
Comment (RobinCsl) [inline apps/landingPage/src/components/info.js]: Is that space necessary?

File: apps/landingPage/src/components/info.js [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+            Maecenas ac nulla enim. Praesent lacinia accumsan congue. Quisque
+            dapibus, eros id vestibulum condimentum, nunc elit lacinia urna,
+            vitae porttitor tortor lorem sit amet erat. Phasellus ut dolor est.
+            Nulla tincidunt at tellus sed posuere. Vestibulum efficitur vehicula
+            est, eget egestas ex varius nec.
+          </Text>
+        </Info>
+        <Info>
+          <Text>
+            {' '}
```
Comment (RobinCsl) [inline apps/landingPage/src/components/info.js]: Idem

File: apps/landingPage/src/components/info.js [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
```
Comment (RobinCsl) [inline apps/landingPage/src/components/info.js]: Idem

File: apps/landingPage/src/components/header.js (hunk 18 above)
Comment (KarinaDavtyan) [reply apps/landingPage/src/components/header.js]: Yes it works only with internal, but anyway I think I ll change this part

File: apps/landingPage/src/components/playground.js [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+export default ({ iframeWidth, iframeHeight }: Props) => (
+  <Playground>
+    <HeadingContainer>
+      <Heading type="title2">It has its own GraphQL server</Heading>
+    </HeadingContainer>
+    <iframe
+      title="GraphQL playground"
+      width={iframeWidth}
+      height={iframeHeight}
+      src="https://kiwi-graphiql.now.sh/"
```
Comment (RobinCsl) [inline apps/landingPage/src/components/playground.js]:
```suggestion
      src="https://p2kwd3i3a8.execute-api.eu-central-1.amazonaws.com/staging/graphql"
```

File: apps/landingPage/src/components/headline.js (hunk 19 above)
Comment (KarinaDavtyan) [reply apps/landingPage/src/components/headline.js]: Sorry I couldnt find the image and temporally take it from tequila page

File: apps/landingPage/src/components/info.js (hunk 20 above)
Comment (KarinaDavtyan) [reply apps/landingPage/src/components/info.js]: It was auto fix, I guess. Will remove it when changing to new design

File: apps/landingPage/src/components/headline.js (hunk 19 above)
Comment (tbergq) [reply apps/landingPage/src/components/headline.js]: You know you can right click on the image in your browser, then click save image as...

Comment (KarinaDavtyan) [reply apps/landingPage/src/components/headline.js]: Didnt think of that😅
I ll add it to the repo in next pr then
